.circleci/docker/build.sh

```diff
@@ -336,6 +336,7 @@ docker build \
        --build-arg "KATEX=${KATEX:-}" \
        --build-arg "ROCM_VERSION=${ROCM_VERSION:-}" \
        --build-arg "PYTORCH_ROCM_ARCH=${PYTORCH_ROCM_ARCH:-gfx900;gfx906}" \
+       --build-arg "USE_DEPLOY=${USE_DEPLOY}" \
```

Comment:
We should delete this line, it turns on deploy builds for everything. We just want it on for the deploy workflow (which we do below in build.sh anyway)